python/ray/train/huggingface/huggingface_trainer.py

```diff
@@ -283,6 +284,27 @@ def __init__(
             resume_from_checkpoint=resume_from_checkpoint,
         )
 
+    @property
+    def train_loop_config(self) -> Dict[str, Any]:
+        trainer_init_config = (
+            self.trainer_init_config.copy() if self.trainer_init_config else {}
+        )
+        if TRAINER_INIT_FN_KEY in trainer_init_config:
+            raise ValueError(
+                f"'{TRAINER_INIT_FN_KEY}' is a reserved key in `trainer_init_config`."
+            )
+        if self.trainer_init_per_worker:
+            trainer_init_config[TRAINER_INIT_FN_KEY] = self.trainer_init_per_worker
+        return trainer_init_config
+
+    @train_loop_config.setter
+    def train_loop_config(self, config):
+        # This disables setting the `train_loop_config` attribute.
+        # The `train_loop_config` for this Trainer depends only on
+        # the `trainer_init_per_worker` and `trainer_init_config` args,
+        # and will be created with the getter above.
+        pass
```

Comment:
Do we call this somewhere? Or is this just if the user calls it? Should we raise a warning or an error if this is called?

Reply:
This is called by us in the super class constructor. This was a hack to get restore to work for the Huggingface/Mosaic trainers, without needing to implement `HuggingFaceTrainer.restore`.

Relates to Jun's comment here: https://github.com/ray-project/ray/pull/31920#discussion_r1087122528